Content.Server/GameObjects/Components/Destructible/Thresholds/Triggers/DamageTypeTrigger.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Shared.Damage;
+using Content.Shared.GameObjects.Components.Damage;
+using Robust.Shared.Serialization;
+
+namespace Content.Server.GameObjects.Components.Destructible.Thresholds.Triggers
+{
+    /// <summary>
+    ///     A trigger that will activate when the amount of damage received
+    ///     of the specified type is above the specified threshold.
+    /// </summary>
+    [Serializable]
+    public class DamageTypeTrigger : IThresholdTrigger
+    {
+        public DamageType? Type { get; set; }
+
+        public int Damage { get; set; }
+
+        public void ExposeData(ObjectSerializer serializer)
+        {
+            serializer.DataField(this, x => x.Type, "type", null);
+            serializer.DataField(this, x => x.Damage, "damage", 0);
+        }
+
+        public bool Reached(IDamageableComponent damageable, DestructibleSystem system)
+        {
+            if (Type == null)
+            {
+                return false;
+            }
+
+            return damageable.TryGetDamage(Type.Value, out var damageReceived) &&
+                   damageReceived >= Damage;
```

Comment:
Won't the DamageTriggers without a type specified not trigger?

Reply:
They won't no, this is mostly because I couldn't think of any other good default behavior.

Reply:
Unless I'm blind it seems like a bunch of the yamls dont have types so are they out of scope?